expression/builtin_json.go

```diff
@@ -651,6 +651,19 @@ func (c *jsonContainsFunctionClass) getFunction(ctx sessionctx.Context, args []E
 	if err := c.verifyArgs(args); err != nil {
 		return nil, err
 	}
+
+	// This should actually do in `c.verifyArgs` but there's no way to inherit a method in Go.
+	evalType := args[0].GetType().EvalType()
+	if evalType != types.ETJson && evalType != types.ETString {
+		return nil, json.ErrInvalidJSONData.GenWithStack(
+			"Invalid data type for JSON data in argument 1 to function json_contains; a JSON string or JSON type is required.")
```

Comment:
Should we upgrade the error type?
I found that MySQL had a special error type for it.
https://github.com/mysql/mysql-server/blob/8.0/share/errmsg-utf8.txt#L7559

Reply:
Sounds reasonable, I just checked the message from MySQL 5.7